pkgs/http/lib/src/base_request.dart

```diff
@@ -89,14 +89,33 @@ abstract class BaseRequest {
   bool _finalized = false;
 
   static final _tokenRE = RegExp(r"^[\w!#%&'*+\-.^`|~]+$");
+
   static String _validateMethod(String method) {
     if (!_tokenRE.hasMatch(method)) {
       throw ArgumentError.value(method, 'method', 'Not a valid method');
     }
     return method;
   }
 
-  BaseRequest(String method, this.url)
+  /// On upload progress callback.
+  ///
+  /// If defined, this callback will be called when the upload progress changes.
+  ///
+  /// In browser, uses XMLHttpRequest's "xhr.upload.onProgress" event.
+  ///
+  /// In IO, uses the yield length of the stream. The total length of the bytes
+  /// yielded by the stream at any given moment is "uploaded" and the total
+  /// length of the stream is "total"
+  ///
+  /// If [onUploadProgress] callback is provided and length is computable,
+  /// [onUploadProgress] will execute for each chunk was sent.
+  ///
+  /// lengthComputable :
+  ///     library.html : xhr.lengthComputable
+  ///     library.io : content-length is provided (MultipartRequest provide)
+  final void Function(int uploaded, int total)? onUploadProgress;
```

Comment:
Hi @Mehmetyaz, thanks for this PR, it is very useful! We are using your fork in our app overriding the dependency of the `http` package until this get deployed in a new version of the package.

Said this, we noted that this signature could be wrong and has the parameters in the wrong order.
In the rest of the code you are calling the `onUploadProgress` callback using the `total` as the first parameter and the `uploaded` bytes as the second one.

Just that. Thanks a lot again.